sencha-workspace/packages/slate-cbl/src/view/student/Dashboard.js

```diff
@@ -47,60 +47,95 @@ Ext.define('Slate.cbl.view.student.Dashboard', {
+                        '<li class="cbl-skill-demo <tpl if="values.Level==0"> cbl-grid-demo-missed</tpl>" <tpl if="DemonstrationID">data-demonstration="{DemonstrationID}"</tpl>>',
+                            '<tpl if="values.Level &gt;= 0">',
+                                '{[values.Level == 0 ? "M" : values.Level]}',
+                            '<tpl else>',
+                                '&nbsp;',
+                            '</tpl>',
+                        '</li>',
+                    '</tpl>',
+                '</ul>',
+                '<div class="cbl-skill-description"><p>{Statement}</p></div>',
 //                '<div class="cbl-skill-complete-indicator cbl-level-{parent.level} is-checked">',
 //                    '<svg class="check-mark-image" width="16" height="16">',
 //                        '<polygon class="check-mark" points="13.824,2.043 5.869,9.997 1.975,6.104 0,8.079 5.922,14.001 15.852,4.07"/>',
 //                    '</svg>',
 //                '</div>',
-        '</li>',
+            '</li>',
         '</tpl>',
-        {
-            getDemonstrationBlocks: function(skill, studentId) {
-                var demonstrationsRequired = skill.DemonstrationsRequired,
-                    blocks = Ext.isArray(skill.demonstrations) ? skill.demonstrations : [];
-
-                blocks = Ext.Array.sort(blocks, function(a, b) {
-                    return a.Level < b.Level ? -1 : 1;
+        
+        {    sortDemonstrations: function sortDemonstrations(demonstrations, limit) {
+                'use strict';
+                demonstrations = Ext.isArray(demonstrations) ? demonstrations : [];
+                
+                var len = demonstrations.length,
+                    x = 0,
+                    i = 0,
+                    demo,
+                    sortedDemonstrations = {},
+                    levelsSorted,
+                    demos,
+                    demosLen,
+                    displayDemonstrations = [];
+                
+                // If no limit is specified, sort all demonstrations
+                limit = isNaN(limit) ? len : limit;
+            
+                // Define functions outside of loops
+                function sortByDate(a, b) {
+                    return a.Demonstrated > b.Demonstrated ? 1 : a.Demonstrated < b.Demonstrated ? -1 : 0;
+                }
+                
+                // Separate demonstrations by level
+                for (x = 0; len > x; x++) {
+                    demo = demonstrations[x];
+                    sortedDemonstrations[demo.Level] = sortedDemonstrations[demo.Level] || [];
+                    sortedDemonstrations[demo.Level].push(demo);
+                }
+                
+                // Sort the levels of demonstrations observed from greatest to least
+                levelsSorted = Object.keys(sortedDemonstrations).sort(function (a, b) {
+                    a = parseInt(a, 10);
+                    b = parseInt(b, 10);
```

Comment:
It's only called once per object key. Even though we used a numeric index, ``Object.keys`` will return them as strings. This allows us to do a numeric comparison. This loop should only run 1-3 times typically (or however many levels were logged on a given score).